pkg/cmd/chat/push/push.go

```diff
@@ -0,0 +1,144 @@
+	"github.com/GetStream/stream-cli/pkg/config"
+	"github.com/GetStream/stream-cli/pkg/utils"
+	"github.com/MakeNowJust/heredoc"
+	"github.com/spf13/cobra"
+)
+
+func NewCmds() []*cobra.Command {
+	return []*cobra.Command{
+		updateCmd(),
+		testCmd(),
+	}
+}
+
+func updateCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "upsert-pushprovider --properties [raw-json]",
+		Short: "Create or updates a push provider",
+		Long: `
+			The "--properties" parameter expects a raw json string that can be
+			unmarshalled into a stream_chat.PushProvider object on the Go SDK side.
+			See the example section.
+			Available properties:
+			type
+			name
+			description
+			disabled_at
+			disabled_reason
+					
+			apn_auth_key
+			apn_key_id
+			apn_team_id
+			apn_topic
+					
+			firebase_credentials
+					
+			huawei_app_id
+			huawei_app_secret
+					
+			xiaomi_package_name
+			xiaomi_app_secret
+		`,
+		Example: heredoc.Doc(`
+			# Setting up an APN push provider
+			$ stream-cli chat upsert-pushprovider --properties "{'type': 'apn', 'name': 'staging', 'apn_auth_key': 'key', 'apn_key_id': 'id', 'apn_topic': 'topic', 'apn_team_id': 'id'}"
+
+			# Setting up a Firebase push provider
+			$ stream-cli chat upsert-pushprovider --properties "{'type': 'firebase', 'name': 'staging', 'firebase_credentials': 'credentials'}"
+
+			# Setting up a Huawei push provider
+			$ stream-cli chat upsert-pushprovider --properties "{'type': 'huawei', 'name': 'staging', 'huawei_app_id': 'id', 'huawei_app_secret': 'secret'}"
+
+			# Setting up a Xiaomi push provider
+			$ stream-cli chat upsert-pushprovider --properties "{'type': 'xiaomi', 'name': 'staging', 'xiaomi_package_name': 'name', 'xiaomi_app_secret': 'secret'}"
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c, err := config.GetConfig(cmd).GetClient(cmd)
+			if err != nil {
+				return err
+			}
+
+			prop, _ := cmd.Flags().GetString("properties")
+
+			var p stream_chat.PushProvider
+			err = json.Unmarshal([]byte(prop), &p)
+			if err != nil {
+				return err
+			}
+
+			_, err = c.UpsertPushProvider(cmd.Context(), &p)
+			if err != nil {
+				return err
+			}
+
+			cmd.Println("Successfully updated push provider.")
+			return nil
+		},
+	}
+
+	fl := cmd.Flags()
+	fl.StringP("properties", "p", "", "[required] Raw json properties to send to the backend")
+	cmd.MarkFlagRequired("properties")
+
+	return cmd
+}
+
+func testCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "test-push --properties [raw-json] --output-format [json|tree]",
+		Short: "Test push notifications",
+		Long: heredoc.Doc(`
+			The "--properties" parameter expects a raw json string that can be
+			unmarshalled into a stream_chat.CheckPushRequest object on the Go SDK side.
+			See the example section.
+			Available properties:
+			message_id
+			apn_template
+			firebase_template
+			firebase_data_template
+			skip_devices
+			push_provider_name
+			push_provider_type
+			user_id
+			user
```

Comment:
I wonder if this should be separate flags (--apn_template) instead of a raw json dump.
Opinion, @ferhatelmas ? Probably traditional flag is better

Reply:
yes, and don't support v1

@driver-devel could you point unnecessary parts?

Reply:
converted it into flags. which are the v1 that I should remove?

Reply:
@peterdeme But we still have --properties (raw json) input for push provider upsert

Reply:
@driver-devel that's way too many properties, so I used raw json there